sdk/eventhub/event-hubs/test/eventProcessor.spec.ts

```diff
@@ -484,5 +484,114 @@ describe("Event Processor", function(): void {
       partitionOwnershipList[0].sequenceNumber!.should.equals(checkpoint.sequenceNumber);
       partitionOwnershipList[0].offset!.should.equals(checkpoint.offset);
     });
+
+    it("should receive events from the checkpoint", async function(): Promise<void> {
```

Comment:
I'm a little confused about this test. It looks like you:
1. Start processor 1
2. Send 100 events
3. Stop processor 1
4. Start processor 2
5. Stop processor 2
6. Check that the first event processor2 checkpoints comes sequentially after the last event processor1 checkpoints.

This seems like it would be a flaky test since it's possible for processor1 to process all events, leaving none left for processor2.

What if instead of checkpointing every event in processor1, you only checkpoint up to the 1st 50 (or some value lower than the total amount of events sent). That way you know even if processor1 sees every event, the checkpointing will stop at 50, and you're guaranteed processor2 will see events.

Reply:
Updated the test case!